server/serverimpl_test.go

```diff
@@ -28,8 +28,8 @@ func startServer(t *testing.T, settings *StartSettings) *server {
 	srv := New(&sharedinternal.NopLogger{})
 	require.NotNil(t, srv)
 	if settings.ListenEndpoint == "" {
-		// Find an avaiable port to listne on.
-		settings.ListenEndpoint = testhelpers.GetAvailableLocalAddress()
+		// allocate an ephemeral port to listne on.
+		settings.ListenEndpoint = "127.0.0.1:0"
```

Comment:
Instead of changing existing tests can we keep existing tests as is and add another simple test that demonstrates the use of 0 port and Addr() method?